applications/plugins/CGALPlugin/CGALPlugin_test/MeshFromImageWithFeatures_test.cpp

```diff
@@ -0,0 +1,222 @@
+#include <sofa/defaulttype/VecTypes.h>
+#include <sofa/simulation/Simulation.h>
+#include <SofaSimulationGraph/DAGSimulation.h>
+#include <sofa/simulation/Node.h>
+#include <SofaBaseMechanics/MechanicalObject.h>
+#include <SofaBaseMechanics/UniformMass.h>
+#include <SceneCreator/SceneCreator.h>
+#include <SofaBoundaryCondition/ConstantForceField.h>
+#include <SofaSimulationCommon/SceneLoaderXML.h>
+
+#define CGALPLUGIN_MESHGENERATIONFROMIMAGE_CPP
+#include <CGALPlugin/MeshGenerationFromImage.h>
+#include <image/ImageTypes.h>
+
+
+namespace sofa{
+namespace {
+
+using namespace modeling;
+using core::objectmodel::New;
+using sofa::simulation::SceneLoaderXML;
+using sofa::core::ExecParams;
+
+
+//template<typename DataTypes>
+//void createUniformMass(simulation::Node::SPtr node, component::container::MechanicalObject<DataTypes>& /*dofs*/)
+//{
+//    node->addObject(New<component::mass::UniformMass<DataTypes, typename DataTypes::Real> >());
+//}
+//
+//template<>
+//void createUniformMass(simulation::Node::SPtr node, component::container::MechanicalObject<defaulttype::Rigid3Types>& /*dofs*/)
+//{
+//    node->addObject(New<component::mass::UniformMass<defaulttype::Rigid3Types, defaulttype::Rigid3Mass> >());
+//}
+//
+//template<>
+//void createUniformMass(simulation::Node::SPtr node, component::container::MechanicalObject<defaulttype::Rigid2Types>& /*dofs*/)
+//{
+//    node->addObject(New<component::mass::UniformMass<defaulttype::Rigid2Types, defaulttype::Rigid2Mass> >());
+//}
+//
+//template <typename _DataTypes>
+struct MeshFromImageWithFeatures_test : public Sofa_test<>
+{
+    sofa::simulation::Node::SPtr root;
+    //typedef _DataTypes DataTypes;
+    //typedef component::projectiveconstraintset::FixedConstraint<DataTypes> FixedConstraint;
+    //typedef component::forcefield::ConstantForceField<DataTypes> ForceField;
+    //typedef component::container::MechanicalObject<DataTypes> MechanicalObject;
+    
+    
+    /*typedef typename MechanicalObject::VecCoord  VecCoord;
+    typedef typename MechanicalObject::Coord  Coord;
+    typedef typename MechanicalObject::VecDeriv  VecDeriv;
+    typedef typename MechanicalObject::Deriv  Deriv;*/
+    //typedef typename DataTypes::Real  Real;
+
+    bool test() //double /*epsilon*/, const std::string &/*integrationScheme */)
+    {
+        cgal::MeshGenerationFromImage< defaulttype::Vec3dTypes, defaulttype::ImageUC >* meshGenerator;
+
+        meshGenerator = dynamic_cast<cgal::MeshGenerationFromImage< defaulttype::Vec3dTypes, defaulttype::ImageUC >*>(root->getObject("generator"));
+
+        helper::ReadAccessor< Data<sofa::defaulttype::Vec3dTypes::VecCoord> > features = meshGenerator->d_features;
+        helper::ReadAccessor< Data<sofa::defaulttype::Vec3dTypes::VecCoord> > nodes = meshGenerator->f_newX0;
+
+        double tolerance = 1e-10;
+        size_t nfts = features.size();
+        bool foundFeatures[nfts];
+        std::cout << "#features to be tested: " << nfts << std::endl;
```

Comment:
std::cout must be replaced with msg_* API. 
You can also use the gtest << feature do provide details about tests: 
EXPECT_EQ(blabhah) << "Provide here the details of failing test you want t add as the number of feature tested: ";